Use common shared dependencies and an explicit plain bootstrap composition.
Origin: EmberBSD (AI-assisted); upstream status: not submitted.

--- a/devel/gettext/Makefile.common
+++ b/devel/gettext/Makefile.common
@@ -25,6 +25,18 @@
 
 TEST_TARGET=	check
 
+# Matching shipped configure files include the patched gnulib macros.
+post-patch:
+	${TOUCH} -r ${WRKSRC}/aclocal.m4 ${WRKSRC}/configure.ac
+	${TOUCH} -r ${WRKSRC}/Makefile.in ${WRKSRC}/Makefile.am
+	${TOUCH} -r ${WRKSRC}/libtextstyle/lib/term-styled-ostream.c ${WRKSRC}/libtextstyle/lib/term-styled-ostream.oo.c
+	${TOUCH} -r ${WRKSRC}/libtextstyle/lib/term-styled-ostream.c ${WRKSRC}/libtextstyle/lib/term-styled-ostream.h
+	${TOUCH} -r ${WRKSRC}/libtextstyle/lib/term-styled-ostream.c ${WRKSRC}/libtextstyle/lib/term_styled_ostream.priv.h
+	${TOUCH} -r ${WRKSRC}/libtextstyle/lib/term-styled-ostream.c ${WRKSRC}/libtextstyle/lib/term_styled_ostream.vt.h
+	${TOUCH} -r ${WRKSRC}/gettext-tools/aclocal.m4 ${WRKSRC}/gettext-tools/gnulib-m4/gnulib-comp.m4
+	${TOUCH} -r ${WRKSRC}/gettext-tools/aclocal.m4 ${WRKSRC}/gettext-tools/configure.ac
+	${TOUCH} -r ${WRKSRC}/libtextstyle/aclocal.m4 ${WRKSRC}/libtextstyle/gnulib-m4/gnulib-comp.m4
+
 .include "../../mk/bsd.fast.prefs.mk"
 
 # There is a bug in the 10.4 SDK that causes struct __darwin_ucontext
--- a/devel/gettext/Makefile
+++ b/devel/gettext/Makefile
@@ -2,7 +2,7 @@
 
 .include "../../devel/gettext/Makefile.common"
 
-PKGREVISION=	1
+PKGREVISION=	2
 COMMENT=	Tools for providing messages in different languages
 
 DEPENDS+=	gettext-m4-${PKGVERSION_NOREV}{,nb[0-9]*}:../../devel/gettext-m4
@@ -12,6 +12,9 @@
 USE_PKGLOCALEDIR=	yes
 SHLIBTOOL_OVERRIDE=	# empty
 CONFIGURE_ARGS+=	--disable-csharp
+CONFIGURE_ARGS+=	--without-included-libxml
+CONFIGURE_ENV+=	EMBER_GETTEXT_EXTERNAL_TEXTSTYLE=yes
+USE_TOOLS+=		pkg-config
 CONFIGURE_ARGS+=	--without-git
 CONFIGURE_ARGS+=	--without-included-gettext
 CONFIGURE_ENV+=		GCJ= ac_cv_prog_GCJ=
@@ -30,6 +33,7 @@
 	cd ${WRKSRC}/gettext-tools/projects && ${MAKE} install DESTDIR=${DESTDIR}
 	cd ${WRKSRC}/gettext-runtime && ${MAKE} install-gettextsrcDATA DESTDIR=${DESTDIR}
 
+.include "../../textproc/libcroco/buildlink3.mk"
 .include "../../devel/gettext-lib/buildlink3.mk"
 .include "../../devel/gettext-asprintf/buildlink3.mk"
 .include "../../mk/bsd.pkg.mk"
--- a/devel/gettext-tools/Makefile
+++ b/devel/gettext-tools/Makefile
@@ -3,8 +3,21 @@
 .include "../../devel/gettext/Makefile.common"
 
 PKGNAME=	${DISTNAME:S/-/-tools-/}
-PKGREVISION=	5
+PKGREVISION=	6
 COMMENT=	Tools for providing messages in different languages
+
+PKG_OPTIONS_VAR=	PKG_OPTIONS.gettext-tools
+PKG_SUPPORTED_OPTIONS=	style
+PKG_SUGGESTED_OPTIONS=	style
+.include "../../mk/bsd.options.mk"
+
+PLIST_VARS+=	style
+.if !empty(PKG_OPTIONS:Mstyle)
+PLIST.style=	yes
+CONFIGURE_ENV+=	EMBER_GETTEXT_EXTERNAL_TEXTSTYLE=yes
+.else
+CONFIGURE_ENV+=	EMBER_GETTEXT_PLAIN_STYLE=yes
+.endif
 
 DISTINFO_FILE=	${.CURDIR}/../../devel/gettext/distinfo
 PATCHDIR=	${.CURDIR}/../../devel/gettext/patches
@@ -19,9 +32,10 @@
 .endif
 USE_PKGLOCALEDIR=	yes
 CONFIGURE_ARGS+=	--disable-csharp
+CONFIGURE_ARGS+=	--without-included-libxml
+USE_TOOLS+=		pkg-config
 CONFIGURE_ARGS+=	--disable-java
 CONFIGURE_ARGS+=	--disable-openmp
-CONFIGURE_ARGS+=	--with-included-libcroco
 CONFIGURE_ARGS+=	--with-xz
 CONFIGURE_ARGS+=	--without-git
 CONFIGURE_ARGS+=	--without-emacs
@@ -76,6 +90,11 @@
 	${INSTALL_MAN} ${WRKSRC}/gettext-tools/man/gettextize.1 \
 		${DESTDIR}${PREFIX}/${PKGMANDIR}/man1
 
+.if !empty(PKG_OPTIONS:Mstyle)
+.include "../../textproc/libcroco/buildlink3.mk"
+.else
+.include "../../textproc/libxml2/buildlink3.mk"
+.endif
 .include "../../devel/gettext-lib/buildlink3.mk"
 .include "../../lang/python/application.mk"
 .include "../../mk/termcap.buildlink3.mk"
--- a/devel/gettext-tools/PLIST
+++ b/devel/gettext-tools/PLIST
@@ -27,15 +27,15 @@
 bin/spit
 bin/xgettext
 include/gettext-po.h
-include/textstyle.h
-include/textstyle/version.h
-include/textstyle/woe32dll.h
+${PLIST.style}include/textstyle.h
+${PLIST.style}include/textstyle/version.h
+${PLIST.style}include/textstyle/woe32dll.h
 info/gettext.info
-info/libtextstyle.info
+${PLIST.style}info/libtextstyle.info
 lib/libgettextlib.la
 lib/libgettextpo.la
 lib/libgettextsrc.la
-lib/libtextstyle.la
+${PLIST.style}lib/libtextstyle.la
 libexec/gettext/cldr-plurals
 libexec/gettext/hostname
 libexec/gettext/project-id
@@ -144,15 +144,15 @@
 share/doc/gettext/textdomain.3.html
 share/doc/gettext/tutorial.html
 share/doc/gettext/xgettext.1.html
-share/doc/libtextstyle/libtextstyle_1.html
-share/doc/libtextstyle/libtextstyle_2.html
-share/doc/libtextstyle/libtextstyle_3.html
-share/doc/libtextstyle/libtextstyle_4.html
-share/doc/libtextstyle/libtextstyle_5.html
-share/doc/libtextstyle/libtextstyle_6.html
-share/doc/libtextstyle/libtextstyle_7.html
-share/doc/libtextstyle/libtextstyle_abt.html
-share/doc/libtextstyle/libtextstyle_toc.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_1.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_2.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_3.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_4.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_5.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_6.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_7.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_abt.html
+${PLIST.style}share/doc/libtextstyle/libtextstyle_toc.html
 share/gettext-${PKGVERSION}/its/docbook.loc
 share/gettext-${PKGVERSION}/its/docbook4.its
 share/gettext-${PKGVERSION}/its/docbook5.its
